AsyncDisplayKit/ASCellNode.mm

```diff
@@ -186,6 +194,38 @@ - (void)setHighlighted:(BOOL)highlighted
   }
 }
 
+- (void)setScrollView:(UIScrollView *)scrollView
+{
+  _scrollView = scrollView;
+
+  if ([scrollView isKindOfClass:[ASTableView class]]) {
+    _scrollViewType.isTableView      = 1;
+    _scrollViewType.isTableNode      = 0;
+    _scrollViewType.isCollectionView = 0;
+    _scrollViewType.isCollectionNode = 0;
+  } else if ([scrollView isKindOfClass:[ASTableNode class]]) {
+    _scrollViewType.isTableView      = 0;
+    _scrollViewType.isTableNode      = 1;
+    _scrollViewType.isCollectionView = 0;
+    _scrollViewType.isCollectionNode = 0;
+  } else if ([scrollView isKindOfClass:[ASCollectionView class]]) {
+    _scrollViewType.isTableView      = 0;
+    _scrollViewType.isTableNode      = 0;
+    _scrollViewType.isCollectionView = 1;
+    _scrollViewType.isCollectionNode = 0;
+  } else if ([scrollView isKindOfClass:[ASCollectionNode class]]) {
+    _scrollViewType.isTableView      = 0;
+    _scrollViewType.isTableNode      = 0;
+    _scrollViewType.isCollectionView = 0;
+    _scrollViewType.isCollectionNode = 1;
+  } else {
+    _scrollViewType.isTableView      = 0;
+    _scrollViewType.isTableNode      = 0;
+    _scrollViewType.isCollectionView = 0;
+    _scrollViewType.isCollectionNode = 0;
+  }
```

Comment:
No need for setting the flags one after another to `0`. Just add at the beginning of the if else block:

```Objc
memset(&_scrollViewType, 0, sizeof(_scrollViewType));
```

 and in the if else set the once to `1`. You can than also remove the else block completely